### PR TITLE
Channel query date parameters

### PR DESCRIPTION
**Pull request by vishalnarkhede**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -821,6 +821,10 @@ export type MuteUserOptions<UserType = UnknownType> = {
 };
 
 export type PaginationOptions = {
+  created_at_after?: string | SeamlessImmutable.ImmutableDate | null;
+  created_at_after_or_equal?: string | SeamlessImmutable.ImmutableDate | null;
+  created_at_before?: string | SeamlessImmutable.ImmutableDate | null;
+  created_at_before_or_equal?: string | SeamlessImmutable.ImmutableDate | null;
```
Comment (vishalnarkhede) [inline src/types.ts:827]: Tests were already added by @thesyncim here - https://github.com/GetStream/stream-chat-js/blob/master/test/integration/test.js#L3348

Comment (vishalnarkhede) [reply src/types.ts:827]: Allowing `SeamlessImmutable.ImmutableDate` for following case:

```js
channel.query({
    messages: {
        created_at_before: channel.lastRead()
    }
})
```

Comment (mahboubii) [reply src/types.ts:827]: if we support `SeamlessImmutable.ImmutableDate` mayeb put `Date` there too?

File: src/types.ts [hunk 2]
```diff
@@ -822,6 +822,10 @@ export type MuteUserOptions<UserType = UnknownType> = {
 };
 
 export type PaginationOptions = {
+  created_at_after?: string | Date | SeamlessImmutable.ImmutableDate | null;
+  created_at_after_or_equal?: string | Date | SeamlessImmutable.ImmutableDate | null;
+  created_at_before?: string | Date | SeamlessImmutable.ImmutableDate | null;
+  created_at_before_or_equal?: string | Date | SeamlessImmutable.ImmutableDate | null;
```
Comment (mahboubii) [inline src/types.ts:828]: we probbaly don't need the null type here
```suggestion
  created_at_after?: string | Date | SeamlessImmutable.ImmutableDate;
  created_at_after_or_equal?: string | Date | SeamlessImmutable.ImmutableDate;
  created_at_before?: string | Date | SeamlessImmutable.ImmutableDate;
  created_at_before_or_equal?: string | Date | SeamlessImmutable.ImmutableDate;
```

**Review by mahboubii — APPROVED** [review]
🌮

**Pull request merged**